controllers/userMigrations.js

```diff
@@ -0,0 +1,61 @@
+const firestore = require("../utils/firestore");
+const userModel = firestore.collection("users");
+const userQuery = require("../models/userMigrations");
+const { getRandomIndex } = require("../utils/helpers");
+const USER_COLORS = 10;
+const MAX_TRANSACTION_WRITES = 499;
+
+/**
+ * Returns the lists of usernames where default colors were added
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const addDefaultColors = async (req, res) => {
+  try {
+    const { usersArr } = await userQuery.returnUsers();
+
+    const batchArray = [];
+    const users = [];
+    batchArray.push(firestore.batch());
+    let operationCounter = 0;
+    let batchIndex = 0;
+    let totalCount = 0;
+
+    for (const user of usersArr) {
+      const colors = user.colors ?? {};
+      if (!user.colors) {
+        const userColorIndex = getRandomIndex(USER_COLORS);
+        colors.color_id = userColorIndex;
+        const docId = userModel.doc(user.id);
+        batchArray[parseInt(batchIndex)].set(docId, { ...user, colors });
+        operationCounter++;
+        totalCount++;
+        users.push(user.username);
+        if (operationCounter === MAX_TRANSACTION_WRITES) {
+          batchArray.push(firestore.batch());
+          batchIndex++;
+          operationCounter = 0;
+        }
+      }
+    }
+    batchArray.forEach(async (batch) => await batch.commit());
```

Comment:
I believe the batch operation will interact with Firestore, and it seems more appropriate to handle it within the models rather than the controller. as we are updating the docs in our DB. I'm still not convinced it should be placed in the controller. but still will rest the decision to you.

Reply:
As per @ankushdharkar - `The model file should only be responsible for interactions with firestore. The manipulation should take place in controller file, if I am not wrong.` Don't you feel the batch operation is a  interaction with firestore @isVivek99 .

Reply:
yes the batch is an interaction with firestore, moving it to models then